lib/std/testing.zig

```diff
@@ -374,6 +374,43 @@ pub const TmpDir = struct {
         self.parent_dir.close();
         self.* = undefined;
     }
+
+    /// Writes program string as zig file into tmp directory
+    /// Caller owns memory
+    ///
+    /// ```
+    /// const progstr = "pub fn main() void {}\n";
+    /// var it = try std.process.argsWithAllocator(std.testing.allocator);
+    /// defer it.deinit(); // no-op unless WASI or Windows
+    /// const testargs = try std.testing.getTestArgs(&it);
+    /// var tmp = std.testing.tmpDir(.{ .no_follow = true }); // ie zig-cache/tmp/8DLgoSEqz593PAEE
+    /// defer tmp.cleanup();
+    /// const zigfile_path = try tmp.writeZigFile(std.testing.allocator, progstr, "bruh");
+    /// defer std.testing.allocator.free(zigfile_path);
+    /// const binary = zigfile_path[0 .. zigfile_path.len - 4]; // '.zig' is 4 characters
+    /// try std.testing.buildExe(testargs.zigexec, zigfile_path, binary);
+    /// ```
+    pub fn writeZigFile(
```

Comment:
I don't think a function called `writeZigFile` belongs in std.testing. If it was a more general-purpose method for writing a temp file  for a test, then I would be OK with it.

Reply:
agreed. changed to writeFile.